sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -682,10 +682,6 @@ object DataSource extends Logging {
                 throw e
               }
           }
-        case _ :: Nil if isUserDefinedDataSource =>
-          // There was DSv1 or DSv2 loaded, but the same name source was found
-          // in user defined data source.
-          throw QueryCompilationErrors.foundMultipleDataSources(provider)
```

Comment:
I think we should still throw an exception if one is found with a short name, but another one is found in Python Data Source.

Reply:
Can we at least document that Scala/JVM short names will be used when the short names are conflicted with Python Data Source?

Reply:
SG

Reply:
Sg! Added usage notes in the python data source user guide.